src/callbacks_step/save_solution_dg.jl

```diff
@@ -78,6 +78,92 @@ function save_solution_file(u, time, dt, timestep,
                             mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equations, dg::DG, cache,
                             solution_callback, element_variables=Dict{Symbol,Any}();
                             system="")
+  if has_parallel()
+    save_solution_file_parallel(u, time, dt, timestep, mesh, equations, dg, cache, solution_callback, element_variables; system=system)
+  else
+    save_solution_file_on_root(u, time, dt, timestep, mesh, equations, dg, cache, solution_callback, element_variables; system=system)
+  end
+end
+
+
+function save_solution_file_parallel(u, time, dt, timestep,
+                                     mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equations, dg::DG, cache,
+                                     solution_callback, element_variables=Dict{Symbol,Any}();
+                                     system="")
+  @unpack output_directory, solution_variables = solution_callback
+
+  # Filename without extension based on current time step
+  if isempty(system)
+    filename = joinpath(output_directory, @sprintf("solution_%06d.h5", timestep))
+  else
+    filename = joinpath(output_directory, @sprintf("solution_%s_%06d.h5", system, timestep))
+  end
```

Comment:
Could you please try to avoid duplicating logic like this and keep it in the main function calling the specialized ones?

Reply:
Done in d0da241faf51cb99b3812eb5b13658d259889f20.